### PR TITLE
Fix `expect(Array).to include(Hash).times` case

### PR DESCRIPTION
**Pull request by ojab**
`#expected` has a [special case](https://github.com/rspec/rspec-expectations/commit/8693ecb75) for single `Hash` and returning this
single `Hash` instead of `[Hash]`, so `expected.first` returns first
`key => value` pair and matcher suddenly fails

### CODE REVIEW

**Comment by ojab** [comment]
Marking as ready for review because of `First-time contributors need a maintainer to approve running workflows. Learn more.`

**Review by pirj — APPROVED** [review]
Nicely done, thank you.
Would you like to add this to the documentation at `features/built_in_matchers/include.feature`, too?

**Comment by ojab** [comment]
Not sure where/what exactly I should put there, added single example to the `hash usage`.

**Comment by ojab** [comment]
oh, and this addition is bogus, it's about matching `actual` hash to some included keys. Apparently my brain doesn't work at this time.

So yeah, should it be just new example in a ruby codeblock or some new test like `Scenario: count usage`?

**Comment by pirj** [comment]
> should it be just new example in a ruby codeblock or some new test like `Scenario: count usage`?

It's a corner case on the "used against a list of hashes" and "when used with expectation counts". I'm on the fence if it needs its own focused scenario or doesn't need one at all since without counts `expect([{ 'a' => 1 }]).to include('a' => 1)` works just fine. Leaving it up to you.

File: features/built_in_matchers/include.feature [hunk 1]
```diff
@@ -108,6 +108,7 @@ Feature: `include` matcher
         it { is_expected.to include(:a) }
         it { is_expected.to include(:b, :a) }
         it { is_expected.to include(:a => 7) }
+        it { is_expected.to include(:b => 5).once }
```
Comment (JonRowe) [inline features/built_in_matchers/include.feature:111]: This doesn't actually pass

**Review by JonRowe — COMMENTED** [review]
This is a reasonable improvement, but the scenario line added actually raises an error, which is why the build is broken, can you take a look?

**Comment by ojab** [comment]
> First-time contributors need a maintainer to approve running workflows.
:okay:
both specs & cucumber is green locally

**Comment by JonRowe** [comment]
Thanks @ojab sorry it took so long to get back to this

**Comment by ojab** [comment]
Oh, I already forgot about this PR :D  
Anyway, not problem at all, I understand that maintainership is a hard & voluntary work, thanks for  doing that.
